### PR TITLE
W-10881227 - allof, oneof and components model, parsing and emission

### PR DESCRIPTION
**Pull request by looseale**

### CODE REVIEW

**Comment by uip-robot-zz** [comment]
Git2Gus App is installed but the `.git2gus/config.json` doesn't exist.

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  private def processAnyMappingField(map: YMap, mapping: AnyMapping, key: String, field: Field)(
+      implicit ctx: DialectContext): Unit = map.key(
+      key,
+      entry => {
+        entry.value.tagType match {
+          case YType.Seq =>
+            try {
+              val nodes = entry.value.as[YSequence].nodes.map(n => AmfScalar(n.as[String], Annotations(n)))
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: extract to method

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+              mapping.set(field, AmfArray(nodes, Annotations(entry.value)), Annotations(entry))
+            } catch {
+              case _: Exception =>
+                ctx.eh.violation(DialectError,
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: [CBI] - extract to method

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      implicit ctx: DialectContext): Unit = map.key(
+      key,
+      entry => {
+        entry.value.tagType match {
+          case YType.Seq =>
+            try {
+              val nodes = entry.value.as[YSequence].nodes.map(n => AmfScalar(n.as[String], Annotations(n)))
+              mapping.set(field, AmfArray(nodes, Annotations(entry.value)), Annotations(entry))
+            } catch {
+              case _: Exception =>
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: Why are we managing exceptions here ? We don't manage exceptions in the rest of parsing code

File: amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala [hunk 4]
```diff
@@ -434,6 +434,18 @@ trait DialectsParsingTest extends DialectTests {
         directory = s"$basePath/conditional"
     )
   }
+
```
Comment (tomsfernandez) [inline amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala:437]: Why comment this ?

File: amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala [hunk 5]
```diff
@@ -474,4 +486,67 @@ trait DialectsParsingTest extends DialectTests {
...
+    cycle(
+        "dialect.yaml",
+        config.golden,
+        Some(Mimes.`application/ld+json`),
+        AMLConfiguration.predefined().withRenderOptions(config.renderOptions.withCompactUris),
+        directory = s"$basePath/allOf"
+    )
+  }
+
+//  multiSourceTest("AllOf facet from JSON-LD", "dialect.%s") { config =>
```
Comment (tomsfernandez) [inline amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala:500]: Why comment out this ?

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

File: amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala (hunk 4 above)
Comment (looseale) [reply amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala:437]: The test is necessary but is not working. This is what I told you about IRI resolution in JSON-LD parsing.

File: amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala (hunk 5 above)
Comment (looseale) [reply amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala:500]: Some of previous comment.

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala (hunk 1 above)
Comment (looseale) [reply amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: Extracted

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala (hunk 2 above)
Comment (looseale) [reply amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: Deleted with the try

File: amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala (hunk 3 above)
Comment (looseale) [reply amf-aml/shared/src/main/scala/amf/aml/internal/parse/dialects/nodemapping/like/AnyMappingParser.scala]: Deleted, the "as" error should be caught by the context

File: amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala (hunk 5 above)
Comment (tomsfernandez) [reply amf-aml/shared/src/test/scala/amf/testing/parsing/DialectsParsingTest.scala:500]: Would probably add the ticket code and a TODO comment

**Review by tomsfernandez — APPROVED** [review]

**Pull request merged**